test/runtime_tests.cc

```diff
@@ -1967,5 +1967,99 @@ namespace detail {
 		REQUIRE(tsk->get_buffer_access_map().get_access_modes(buff_id).count(cl::sycl::access_mode::discard_read_write) == 1);
 	}
 
+	TEST_CASE_METHOD(buffer_manager_fixture, "accessor multi-subscriptor for 3D", "[accessor]") {
+		size_t N = 2;
+		size_t N2 = 3;
+		size_t N3 = 4;
+		cl::sycl::range range = cl::sycl::range<3>(N, N2, N3);
+		auto& bm = get_buffer_manager();
+		auto bid = bm.register_buffer<cl::sycl::id<3>, 3>(range);
+		access_target tgt = access_target::HOST;
+
+		SECTION("DEVICE ACCESSOR") {
+			auto& q = get_device_queue();
+			auto sr = subrange<3>({}, range_cast<3>(range));
+			live_pass_device_handler cgh(nullptr, sr, q);
+
+			auto acc3d = get_device_accessor<cl::sycl::id<3>, 3, cl::sycl::access::mode::read_write>(cgh, bid, range, {});
+			cgh.parallel_for<class test_3D_subscriptor>(range, [=](cl::sycl::item<3> item) {
+				size_t i = item[0];
+				size_t j = item[1];
+				size_t k = item[2];
+				acc3d[i][j][k] = item.get_id();
+			});
+			cgh.get_submission_event().wait();
+		}
+
+		SECTION("HOST ACCESSOR") {
+			distr_queue q;
+			q.submit([=](handler& cgh) {
+				auto acc3d = get_host_accessor<cl::sycl::id<3>, 3, cl::sycl::access::mode::read_write>(bid, range, {});
+				cgh.host_task(on_master_node, [=] {
+					for(size_t k = 0; k < N3; k++) {
+						for(size_t j = 0; j < N2; j++) {
+							for(size_t i = 0; i < N; i++) {
+								acc3d[i][j][k] = {i, j, k};
+							}
+						}
+					}
+				});
+			});
+			q.slow_full_sync();
+		}
+
+		bool acc_check = buffer_reduce<cl::sycl::id<3>, 3, class check_chain_subscriptor_host>(
+		    bid, tgt, range, {}, true, [](cl::sycl::id<3> idx, bool current, cl::sycl::id<3> value) { return current && value == idx; });
+
+		REQUIRE(acc_check);
+	}
+
+	TEST_CASE_METHOD(buffer_manager_fixture, "accessor multi-subscriptor for 2D", "[accessor]") {
+		size_t N = 2;
+		size_t N2 = 3;
+		cl::sycl::range range3 = cl::sycl::range<3>(N, N2, 1);
+		cl::sycl::range range2 = cl::sycl::range<2>(N, N2);
+		auto& bm = get_buffer_manager();
+		auto bid = bm.register_buffer<cl::sycl::id<2>, 2>(range3);
+		access_target tgt = access_target::HOST;
+
+		SECTION("DEVICE ACCESSOR") {
```

Comment:
Why all-caps?

Reply:
To add to that; try to have test case names + sections form (somewhat) complete sentences, so we get nice error messages like this:
![image](https://user-images.githubusercontent.com/791348/122186428-531cb400-ce8e-11eb-88dc-909a7a86bf44.png)